src/main/java/org/folio/circulation/domain/Item.java

```diff
@@ -423,4 +423,17 @@ public Item withLoanType(JsonObject newLoanTypeRepresentation) {
       this.primaryServicePoint,
       newLoanTypeRepresentation);
   }
+
+  public Item withLastCheckIn(LastCheckIn lastCheckIn) {
+    Item item = new Item(
+      itemRepresentation.put(ItemProperties.LASTCHECKIN, lastCheckIn.toJson()),
```

Comment:
I would prefer the domain kept a reference to this class and the representation changed at the point of storage, rather than immediately manipulating the JSON.

The more JSON manipulation can be moved to the edges the better from my perspective.

Reply:
Done.

Reply:
Note: tried to do it, but the coupling to items's JSON representation, which is used in all domain classes is too high right now and it will require great care and a LOT of testing avoid introducting new bugs.

Reply:
I appreciate you trying. Can you help me understand why there is coupling to this new information, nothing should be using it at this point?

Reply:
The previous implementation is in previous commit, fff380b55ab0f5b1ccac2a328900d7562a67ba35, createItemStorageRepresentation() in ItemSummaryRepresentation class. 
But it appeared to introduce bugs, so had to be re-written to use item json directly.
Are we moving away from having json representations in domain classes anytime soon?

Reply:
> The previous implementation is in previous commit, [fff380b](https://github.com/folio-org/mod-circulation/commit/fff380b55ab0f5b1ccac2a328900d7562a67ba35), createItemStorageRepresentation() in ItemSummaryRepresentation class.
> But it appeared to introduce bugs, so had to be re-written to use item json directly.

I don’t think I follow. The current latest version of this branch appears to not use the JSON directly. Does that mean you found the cause of the issues?

> Are we moving away from having json representations in domain classes anytime soon?

It has been my intent to start doing that for a few months. However it is unlikely to be a quick process. I doubt it could be done in one change.

My focus is on reducing new code introducing more of this debt. I don’t have time to start addressing most of it.